src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsTests.java

```diff
@@ -699,6 +707,55 @@ public void run() {
         }
     }
 
+    @Test
+    public void testNodeRejectsClusterStateWithWrongMasterNode() throws Exception {
```

Comment:
I think this test can be moved somewhere else and use the single node base class? this doesn't really have a disruption in it and will be slow...